hpx/runtime/threads/policies/lockfree_queue_backends.hpp

```diff
@@ -63,6 +63,8 @@ namespace hpx { namespace threads { namespace policies {
         bool pop(reference val, bool steal = true)
         {
 #if defined(HPX_HAVE_CXX11_STD_ATOMIC_128BIT)
+            if (steal)
+                return queue_.pop_left(val);
```

Comment:
Note that this makes `lockfree_fifo_backend` *exactly* the same as `lockfree_abp_fifo_backend` below (modulo the handling of 128 bit atomics). Same thing with `lockfree_lifo_backend`.

Reply:
Fair point. We might be able to get rid of the abp backends... I think it is important to have the queue stealing from the opposite end in any case.